module/VuFindConsole/src/VuFindConsole/Controller/ScheduledSearchController.php

```diff
@@ -0,0 +1,467 @@
+ * @link     https://vufind.org/wiki/development:plugins:controllers Wiki
+ */
+namespace VuFindConsole\Controller;
+
+use Zend\Console\Console;
+use Zend\ServiceManager\ServiceLocatorInterface;
+
+/**
+ * CLI Controller Module (scheduled search tools)
+ *
+ * @category VuFind
+ * @package  Controller
+ * @author   Samuli Sillanpää <samuli.sillanpaa@helsinki.fi>
+ * @author   Ere Maijala <ere.maijala@helsinki.fi>
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development:plugins:controllers Wiki
+ */
+class ScheduledSearchController extends AbstractBase
+    implements \VuFind\I18n\Translator\TranslatorAwareInterface
+{
+    use \VuFind\I18n\Translator\TranslatorAwareTrait;
+    use \VuFind\I18n\Translator\LanguageInitializerTrait;
+
+    /**
+     * Useful date format value
+     *
+     * @var string
+     */
+    protected $iso8601 = 'Y-m-d\TH:i:s\Z';
+
+    /**
+     * HMAC generator
+     *
+     * @var \VuFind\Crypt\HMAC
+     */
+    protected $hmac;
+
+    /**
+     * View renderer
+     *
+     * @var \Zend\View\Renderer\PhpRenderer
+     */
+    protected $renderer;
+
+    /**
+     * URL helper
+     *
+     * @var \Zend\View\Helper\Url
+     */
+    protected $urlHelper;
+
+    /**
+     * Search results plugin manager
+     *
+     * @var \VuFind\Search\Results\PluginManager
+     */
+    protected $resultsManager;
+
+    /**
+     * Configured schedule options
+     *
+     * @var array
+     */
+    protected $scheduleOptions;
+
+    /**
+     * Top-level VuFind configuration
+     *
+     * @var \Zend\Config\Config
+     */
+    protected $mainConfig;
+
+    /**
+     * Number of results to retrieve when performing searches
+     *
+     * @var int
+     */
+    protected $limit = 50;
+
+    /**
+     * Mail service
+     *
+     * @var \VuFind\Mailer\Mailer
+     */
+    protected $mailer;
+
+    /**
+     * Constructor
+     *
+     * @param ServiceLocatorInterface $sm Service locator
+     */
+    public function __construct(ServiceLocatorInterface $sm)
+    {
+        parent::__construct($sm);
+
+        $this->hmac = $sm->get(\VuFind\Crypt\HMAC::class);
+        $this->renderer = $sm->get('ViewRenderer');
+        $this->urlHelper = $this->renderer->plugin('url');
+        $this->resultsManager = $sm->get(
+            \VuFind\Search\Results\PluginManager::class
+        );
+        $this->scheduleOptions = $sm
+            ->get(\VuFind\Search\History::class)
+            ->getScheduleOptions();
+        $this->mainConfig = $sm->get(\VuFind\Config\PluginManager::class)
+            ->get('config');
+        $this->mailer = $sm->get(\VuFind\Mailer\Mailer::class);
+    }
+
+    /**
+     * Send notifications.
+     *
+     * @return \Zend\Console\Response
+     */
+    public function notifyAction()
+    {
+        $this->processViewAlerts();
```

Comment:
Should output a startup message, otherwise all the other messages are indented too much.

Reply:
Since we already have a message for "Processing x searches," I don't think we need a different startup message; instead, I've de-indented all of the msg() outputs by four spaces. Let me know if you have any concerns about this!